### DISCUSSION TITLE
custom event trigger from native JS in document.body click event, is it possible?

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#4124] [1. Help] [answered]
![image](https://github.com/alpinejs/alpine/assets/9047436/91e0aea6-42c5-433d-be2a-3134306bed03)

![image](https://github.com/alpinejs/alpine/assets/9047436/556ae597-da11-4892-837a-a3fbc8b732e0)

Hi,

I saw $dispatch example, and I knew that is inside alpinejs global scope, but if I want a document.body outside of alpinejs to trigger it, is it possible? If yes, how to do?

Thanks
Eric Xin

**Comment by SimoTod** [comment]
alpine just uses standard browser events: https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events

**Comment by ericxin1982** [comment]
But I use document.dispatchEvent('response') or document.dispatchevent(response), console mentioned the which is not defined.

**Reply by SimoTod** [reply]
Are you referring to the first screenshot? Your listener is on the window object, you dispatch on document. Hard to say without seeing the code but I suppose your event doesn't have the bubbles flag set to true: https://developer.mozilla.org/en-US/docs/Web/API/Event/Event

**Comment by ekwoka** [comment]
```
document.dispatchEvent('response')
```

You can't actually do that.

`dispatchEvent` requires being called with an `Event` not a string.

Then set bubble true, and yo ucan dispatch it from anywhere.

In Alpine's case with $dispatch, the event will bubble, so can you not just call it from Alpine elsewhere? or?

**Comment by ericxin1982** [comment]
![image](https://github.com/alpinejs/alpine/assets/9047436/fe4e3621-4357-471e-87d5-56767fa510f5)

![image](https://github.com/alpinejs/alpine/assets/9047436/231dfe1b-98f5-485f-afe3-f3c91a92ac01)

I finally put a empty x-data in body, and bind click event to dispatch custom event defined inside AlpineJS env, that works. 
I forgot I can create empty x-data object in body, that is also feasible to me. It is ok to go by this way right?

**Reply by ekwoka** [reply]
Sure. It just will make the initialization quite a bit more work, but it's not a killer thing.

**Reply by ericxin1982** [reply]
OK, if I do not use alpinejs dispatch to invoke a custom event, what is other way existing?

**Reply by ericxin1982** [reply]
https://laracasts.com/series/alpine-essentials/episodes/8
I found answer. That is more concise.

**Reply by ekwoka** [reply]
Yeah. That's normal JavaScript event apis. Great👍

**Comment by ericxin1982** [comment]
By the way, I just saw $dispatch used inside template, how about using inside x-data related object, since if I use $dispatch only, console warned me $dispatch is not defined, so I give up and used it in template, but I still have chance to use dispatch inside function within x-data object. how to solve?

**Reply by ekwoka** [reply, marked as answer]
You can just use `this.$dispatch`
